code-challenges-401/graph/__tests__/graph.test.js

```diff
@@ -0,0 +1,22 @@
+const graphMain = require('../graph.js');
+const Vertex = graphMain.vertex;
+const Graph = graphMain.graph;
+
+describe('testing graph and methods', () => {
+    it('should add a new vertex to graph', () => {
+        let test = new Graph();
+        let vertece = new Vertex('vertece');
+
+        test.addVertex(vertece);
+        expect(test.adjacencyList.get(vertece)).toBeTruthy();
+    });
+    it('should add new edge to two vertece', () => {
+        let test = new Graph();
+        let vertece = new Vertex('vertece');
+        let verteceTwo = new Vertex('verteceTwo');
+        
+        test.addVertex(vertece); 
+        test.addVertex(verteceTwo);
+        expect(test.adjacencyList.get(vertece)[0].vertex.value).toEqual(0);
+    })
+})
```

Comment:
Code looks really great Nick, I would love to see more tests added:
- A collection of all nodes can be properly retrieved from the graph
- All appropriate neighbors can be retrieved from the graph
- Neighbors are returned with the weight between nodes included
- The proper size is returned, representing the number of nodes in the graph
- A graph with only one node and edge can be properly returned
- An empty graph properly returns null